.github/workflows/action_tests.yaml

```diff
@@ -145,3 +145,61 @@ jobs:
           cd local-action
           npm install
           ./action_tests/assert.js all-hold-the-line-no-upload-id
+
+  pull_request_autofix:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: local-action
+
+      - name: Set up test
+        shell: bash
+        run: |
+          ./local-action/action_tests/setup.sh src-repo repo-under-test
+          cd repo-under-test
+          git checkout feature-branch
+          echo "EXPECTED_UPSTREAM=$(git rev-parse feature-branch^1)" >>$GITHUB_ENV
+
+      - name: Run trunk-action
+        shell: bash
+        run: |
+          cd repo-under-test
+          git checkout feature-branch
+          ../local-action/pull_request.sh
+        env:
+          INPUT_ARGUMENTS: ""
+          INPUT_CHECK_RUN_ID: ""
+          INPUT_DEBUG: ""
+          INPUT_LABEL: ""
+          TRUNK_PATH: ../local-action/action_tests/stub.js
+          INPUT_GITHUB_REF_NAME: feature-branch
+          GITHUB_EVENT_PULL_REQUEST_NUMBER: ""
+          GITHUB_EVENT_PULL_REQUEST_BASE_SHA: ${{ env.EXPECTED_UPSTREAM }}
+          GITHUB_EVENT_PULL_REQUEST_HEAD_SHA: ""
+          GITHUB_EVENT_PULL_REQUEST_HEAD_REPO_FORK: ""
+          INPUT_SAVE_ANNOTATIONS: ""
+          INPUT_AUTOFIX_AND_PUSH: true
+
+      - name: Assert CLI calls
+        shell: bash
+        run: |
+          cd local-action
+          npm install
+          ./action_tests/assert.js pull-request-autofix
+
+      - name: Assert state of feature-branch
+        shell: bash
+        run: |
+          cd repo-under-test
+          git fetch origin
+          actual_commit_count=$(git rev-list --count feature-branch)
+          expected_commit_count=3
+          if [ "$actual_commit_count" -eq "$expected_commit_count" ]; then
+            echo "The branch has $expected_commit_count commits as expected."
+            exit 0
+          else
+            echo "The branch does not have $expected_commit_count commits."
+            exit 1
+          fi
```

Comment:
let's base this assertion on the latest commit message for `origin/feature-branch`